server/lib/voice_parser.py

```
@@ -0,0 +1,163 @@
+            contents=[types.Part.from_text(text=PROMPT.format(transcript=cleaned))],
+            config=types.GenerateContentConfig(
+                response_mime_type="application/json",
+                response_schema=VoiceParseResult,
+            ),
+        )
+    except Exception as e:
+        # Normalize SDK/network/auth errors into RuntimeError so the router
+        # maps them to a 502 instead of leaking through as an unhandled 500.
+        # This honors the documented contract above.
+        logger.exception("Voice parser provider call failed")
+        raise RuntimeError("Voice parsing service is temporarily unavailable.") from e
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)